### PR TITLE
[Merged by Bors] - feat: `Matrix.abs_det_submatrix_equiv_equiv`

### PR DESCRIPTION
**Pull request by madvorak**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean [hunk 1]
```diff
@@ -222,6 +222,18 @@ theorem det_submatrix_equiv_self (e : n ≃ m) (A : Matrix m m R) :
...
+determinant. -/
+@[simp]
+theorem abs_det_submatrix_equiv_equiv {R : Type*} [LinearOrderedCommRing R]
+    (e₁ e₂ : n ≃ m) (A : Matrix m m R) :
+    |(A.submatrix e₁ e₂).det| = |A.det| := by
+  have hee : e₂ = e₁.trans (e₁.symm.trans e₂) := by ext; simp
+  rw [hee]
+  show |((A.submatrix id (e₁.symm.trans e₂)).submatrix e₁ e₁).det| = |A.det|
+  rw [Matrix.det_submatrix_equiv_self, Matrix.det_permute']
+  cases' Int.units_eq_one_or (sign (e₁.symm.trans e₂)) with he he <;> rw [he] <;> simp
```
Comment (Vierkantor) [inline Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean:235]: I would have hoped that we could do this as one rewrite, by rewriting with `abs_mul` and then saying the cast of `Units Int` always has absolute value 1. I did find `AbsoluteValue.map_units_intCast`, but that's for bundled absolute value operators.

Could you extract a lemma saying `|((u : Int) : R)| = 1` for `u : Units Int`?

Comment (madvorak) [reply Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean:235]: Do you want to eliminate line 234 as well, or just simplify line 235 by this change?

Comment (Vierkantor) [reply Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean:235]: I mean that we could make line 234 into a oneliner along the lines of:
```lean
  rw [Matrix.det_submatrix_equiv_self, Matrix.det_permute', abs_mul, abs_units_intCast, one_mul]
```

**Comment by madvorak** [comment]
Which file should `abs_unit_coe` go to?

**Comment by Vierkantor** [comment]
> Which file should `abs_unit_coe` go to?

According to `set_option linter.minImports true in lemma abs_unit_coe`, we want a file that imports `Mathlib.Algebra.Order.Ring.Abs` and `Mathlib.Algebra.Ring.Int.Units`. I think it would be fine to add an import of `Mathlib.Algebra.Ring.Int.Units` to `Mathlib.Algebra.Order.Ring.Int` and put it in the latter file.

**Comment by madvorak** [comment]
> > Which file should `abs_unit_coe` go to?
> 
> According to `set_option linter.minImports true in lemma abs_unit_coe`, we want a file that imports `Mathlib.Algebra.Order.Ring.Abs` and `Mathlib.Algebra.Ring.Int.Units`. I think it would be fine to add an import of `Mathlib.Algebra.Ring.Int.Units` to `Mathlib.Algebra.Order.Ring.Int` and put it in the latter file.

Unfortunately, this environment does not know that `|1| = 1` holds.

**Comment by madvorak** [comment]
Importing `Mathlib.Algebra.Order.Ring.Abs` makes a cycle.

**Comment by madvorak** [comment]
Can it go into `Mathlib.Algebra.Order.Ring.Abs` instead?

**Comment by madvorak** [comment]
Please also review the lemma name.

File: Mathlib/Algebra/Order/Ring/Abs.lean [hunk 2]
```diff
@@ -151,6 +152,9 @@ theorem abs_sub_sq (a b : α) : |a - b| * |a - b| = a * a + b * b - (1 + 1) * a
   simp only [mul_add, add_comm, add_left_comm, mul_comm, sub_eq_add_neg, mul_one, mul_neg,
     neg_add_rev, neg_neg, add_assoc]
 
+lemma abs_unit_coe (a : ℤˣ) : |((a : ℤ) : α)| = 1 := by
```
Comment (Vierkantor) [inline Mathlib/Algebra/Order/Ring/Abs.lean:155]: Good point about the lemma name, I forgot to add that suggestion!

How about:
```suggestion
lemma abs_unit_intCast (a : ℤˣ) : |((a : ℤ) : α)| = 1 := by
```

**Review by Vierkantor — APPROVED** [review]
Okay, this is looking good to me! Please wait for the build to complete successfully and then reply with `bors merge` to merge this pull request.

bors d+

**Comment by madvorak** [comment]
bors merge

**Pull request closed without merging**